benchmarking/PureSynthetic2D/ensemble_scikit.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC
+from typing import Any, Dict, Generic, List, Optional, Type, TypeVar, Union, cast
+
+import numpy as np
+import sklearn
+import torch
+from sklearn.metrics import explained_variance_score, mean_squared_error, r2_score
+from sklearn.model_selection import KFold
+
+from pyrelational.model_managers import ModelManager
+
+
+class EnsembleScikit(ModelManager[sklearn.base.BaseEstimator, List[sklearn.base.BaseEstimator]]):
+    """Custom ModelManager for an ensemble of K-fold SVM regressors, as found in query-by-commitee
+    strategies. Can easily be adapted to any other regressor
+
+    Args:
+    model_class (sklearn estimator): Estimator that should be ensembled (e.g. MLPRegressor)
+    num_estimators (int): number of estimators in the ensemble
+    model_config (dict): dictionary containing any model_class specific arguments
+    trainer_config (dict): dictionary containing any taining specific arguments
+    """
+
+    def __init__(
+        self,
+        model_class: sklearn.base.BaseEstimator,
+        num_estimators: int,
+        model_config: Dict[str, Any],
+        trainer_config: Dict[str, Any],
+    ):
+        super(EnsembleScikit, self).__init__(model_class, model_config, trainer_config)
+        self.model_config = model_config
+        self.trainer_config = trainer_config
+        self.num_estimators = num_estimators
+
+    def train(
+        self,
+        train_loader: torch.utils.data.DataLoader[Any],
+        validation_loader: Optional[torch.utils.data.DataLoader[Any]] = None,
+    ) -> None:
+        """
+        Args:
+        train_loader (torch.utils.data.DataLoader): A torch dataloader with a numpy compatible collate function
+            you can also just adapt this to something else.
+        """
+        train_x, train_y = next(iter(train_loader))  # assumes dataloader returns full set of available observations
+        estimators: List[Any] = []
+
+        k = self.num_estimators
+        kf = KFold(n_splits=k)
+        from joblib import Parallel, delayed
```

Comment:
can this be moved to top of file instead? could also used this in inference maybe?